<a href="https://colab.research.google.com/github/vinnybun/FIP-GENERATIVE-AI-AND-DATA-SCIENCE-PROJECTS/blob/main/deliverable_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### FIP GENERATIVE ARTIFICIAL INTELLIGENCE AND DATA SCIECNCE (GAIDS) DELIVERABLE 3

This week our focus waa on foundational statistics and probabilities. Hence the deliverable for the topic is ;

Analyze a small, publicly available dataset using Python's NumPy or Pandas libraries.                                                                                                
1. Explore the data:

Print the first few rows using data.head().
Check the data types of each column using data.info().
Get summary statistics (mean, median, standard deviation, etc.) using data.describe().

2. Calculate basic statistics:

Calculate the mean, median, mode, and standard deviation for each numerical feature.
Calculate the correlation between different features using data.corr().


The dataset used for this task was gotten from kaggle: [Amazon Sales dataset](https://www.kaggle.com/datasets/aliiihussain/amazon-sales-dataset)

In [ ]:
#importing necessary libraries
import pandas as pd
import numpy as np
import kagglehub
import os


In [ ]:

# Download latest version of the dataset
path = kagglehub.dataset_download("aliiihussain/amazon-sales-dataset")

print("Path to dataset files:", path)



Using Colab cache for faster access to the 'amazon-sales-dataset' dataset.
Path to dataset files: /kaggle/input/amazon-sales-dataset


In [ ]:
#define our path
path = "/root/.cache/kagglehub/datasets/aliiihussain/amazon-sales-dataset/versions/1"

#list files in the directory to list files in csv
files = os.listdir(path)
print(f"Files in the directory : {files}")

Files in the directory : ['amazon_sales_dataset.csv']


In [ ]:
#load into a pandas dataframe
df = pd.read_csv(os.path.join(path, "amazon_sales_dataset.csv"))

After successfully loading our downloaded dataset from kaggle into a variable "df" it is safe to explore the data and perform some basic statistics on the dataset

In [ ]:
#The first 10 few rows of the dataset
df.head(10)


order_id  order_date  product_id product_category   price  \
0         1  2022-04-13        2637            Books  128.75   
1         2  2023-03-12        2300          Fashion  302.60   
2         3  2022-09-28        3670           Sports  495.80   
3         4  2022-04-17        2522            Books  371.95   
4         5  2022-03-13        1717           Beauty  201.68   
5         6  2023-12-02        1777           Sports  385.17   
6         7  2022-01-21        4068            Books   15.78   
7         8  2023-09-07        3537          Fashion  111.59   
8         9  2022-05-02        3262            Books  373.62   
9        10  2023-04-12        4174           Sports  134.81   

   discount_percent  quantity_sold customer_region    payment_method  rating  \
0                10              4   North America               UPI     3.5   
1                20              5            Asia       Credit Card     3.7   
2                20              2          Europe               UPI     4.4   
3                15              4     Middle East               UPI     5.0   
4                 0              4     Middle East               UPI     4.6   
5                30              3            Asia            Wallet     3.8   
6                 0              5            Asia               UPI     1.6   
7                30              4     Middle East               UPI     3.4   
8                 0              4            Asia            Wallet     2.8   
9                15              2     Middle East  Cash on Delivery     2.0   

   review_count  discounted_price  total_revenue  
0           443            115.88         463.52  
1           475            242.08        1210.40  
2           183            396.64         793.28  
3           212            316.16        1264.64  
4           308            201.68         806.72  
5           489            269.62         808.86  
6           415             15.78          78.90  
7           143             78.11         312.44  
8           497            373.62        1494.48  
9            25            114.59         229.18

In [ ]:
#check the information about the dataset
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   order_id          50000 non-null  int64  
 1   order_date        50000 non-null  object 
 2   product_id        50000 non-null  int64  
 3   product_category  50000 non-null  object 
 4   price             50000 non-null  float64
 5   discount_percent  50000 non-null  int64  
 6   quantity_sold     50000 non-null  int64  
 7   customer_region   50000 non-null  object 
 8   payment_method    50000 non-null  object 
 9   rating            50000 non-null  float64
 10  review_count      50000 non-null  int64  
 11  discounted_price  50000 non-null  float64
 12  total_revenue     50000 non-null  float64
dtypes: float64(4), int64(5), object(4)
memory usage: 5.0+ MB
None


In [ ]:
#summary statistics of the numerical data in the dataset
df.describe()

order_id    product_id         price  discount_percent  \
count  50000.000000  50000.000000  50000.000000      50000.000000   
mean   25000.500000   2986.848740    252.507260         13.340700   
std    14433.901067   1156.374535    143.025544          9.850694   
min        1.000000   1000.000000      5.010000          0.000000   
25%    12500.750000   1983.000000    127.840000          5.000000   
50%    25000.500000   2983.000000    252.970000         10.000000   
75%    37500.250000   3989.000000    376.335000         20.000000   
max    50000.000000   4999.000000    499.990000         30.000000   

       quantity_sold        rating  review_count  discounted_price  \
count   50000.000000  50000.000000  50000.000000      50000.000000   
mean        2.999400      2.996316    249.329280        218.886566   
std         1.415401      1.154295    144.251981        127.317681   
min         1.000000      1.000000      0.000000          3.530000   
25%         2.000000      2.000000    125.000000        109.680000   
50%         3.000000      3.000000    250.000000        215.805000   
75%         4.000000      4.000000    374.000000        322.702500   
max         5.000000      5.000000    499.000000        499.910000   

       total_revenue  
count   50000.000000  
mean      657.331475  
std       526.223968  
min         3.840000  
25%       240.320000  
50%       505.410000  
75%       968.970000  
max      2499.550000

After the Data Exploration we discovered that the dataset has 13 columns of which 9 are numerical (4 columns are of float datatype, 5 columns are integer datatype) and 4 columns are categorical/ non-numerical (object datatype).

The column "order_id" appears to be the unique identifier of the dataset of which 5000 non-missing unique observations were recorded.

The dataset appears to be symmetrical and not skewed as the mean is approximately equal to the medians across all columns.

After the Data Exploration we were tasked to to calculate some basic statistics in the dataset.

firstly we calculate the mean, median and standard deviation of the numerical columns

In [ ]:
#BASIC STAISTICS OF THE NUMERICAL COLUMN

#selecting the numerical columns
numerical_columns = df.select_dtypes(["int", "float"]).columns
print(f"The numerical columns are : {numerical_columns}")

#calculate the mean, median, standard deviation of the numerical columns
for col in numerical_columns:
  #calculate the mean
  mean = df[col].mean()
  print("\n",f"This is the mean of {col}: {mean}", "\n")

  #calculate the median
  median = df[col].median()
  print(f"The median for the {col} column is {median}", "\n")

  #calculate the standard deviation
  std_dev = df[col].std()
  print(f"The standard Deviation of the {col} column is {std_dev}", "\n")

The numerical columns are : Index(['order_id', 'product_id', 'price', 'discount_percent', 'quantity_sold',
       'rating', 'review_count', 'discounted_price', 'total_revenue'],
      dtype='object')

 This is the mean of order_id: 25000.5 

The median for the order_id column is 25000.5 

The standard Deviation of the order_id column is 14433.90106658626 


 This is the mean of product_id: 2986.84874 

The median for the product_id column is 2983.0 

The standard Deviation of the product_id column is 1156.3745351492848 


 This is the mean of price: 252.50726 

The median for the price column is 252.97 

The standard Deviation of the price column is 143.0255444861365 


 This is the mean of discount_percent: 13.3407 

The median for the discount_percent column is 10.0 

The standard Deviation of the discount_percent column is 9.850693591481008 


 This is the mean of quantity_sold: 2.9994 

The median for the quantity_sold column is 3.0 

The standard Deviation of the quantity_sold col

Looking at the mean, median and standard deviation of the numerical columns, it clearly depicts the same figures as gotten in the summary statistics of the dataframe. Next we get the correlation of different features in the dataset.
I would be using a correlation matrix to get the relationship and how strongly correlated the different features of the dataset are to each other.

In [ ]:
# create a correlation matrix
correlation_matrix = df.corr(numeric_only=True)

correlation_matrix

order_id  product_id     price  discount_percent  \
order_id          1.000000   -0.004990  0.008633         -0.006286   
product_id       -0.004990    1.000000 -0.001639         -0.001623   
price             0.008633   -0.001639  1.000000         -0.004654   
discount_percent -0.006286   -0.001623 -0.004654          1.000000   
quantity_sold    -0.009524   -0.002436  0.004405          0.001399   
rating            0.004293   -0.008535  0.006386          0.003715   
review_count      0.006928   -0.009896  0.001282          0.005193   
discounted_price  0.010069   -0.001447  0.974717         -0.198713   
total_revenue     0.002340   -0.004337  0.711094         -0.144392   

                  quantity_sold    rating  review_count  discounted_price  \
order_id              -0.009524  0.004293      0.006928          0.010069   
product_id            -0.002436 -0.008535     -0.009896         -0.001447   
price                  0.004405  0.006386      0.001282          0.974717   
discount_percent       0.001399  0.003715      0.005193         -0.198713   
quantity_sold          1.000000 -0.002879      0.000717          0.004457   
rating                -0.002879  1.000000     -0.002895          0.006366   
review_count           0.000717 -0.002895      1.000000          0.000829   
discounted_price       0.004457  0.006366      0.000829          1.000000   
total_revenue          0.594404  0.001804      0.002208          0.729694   

                  total_revenue  
order_id               0.002340  
product_id            -0.004337  
price                  0.711094  
discount_percent      -0.144392  
quantity_sold          0.594404  
rating                 0.001804  
review_count           0.002208  
discounted_price       0.729694  
total_revenue          1.000000

Usually I would remove the columns : order_id and product_id beacause they are unique identifiers in the dataframe.

From the correlation matrix it can be seen that the columns involving pricing has a strong correlation in the data
where disconted price and price exhibits the strongest correlation of 0.97